lib/src/rules/no_adjacent_strings_in_list.dart

```diff
@@ -55,6 +58,20 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   _Visitor(this.rule);
 
+  @override
+  void visitForElement(ForElement node) {
+    if (node.body is AdjacentStrings) {
+      rule.reportLint(node.body);
+    }
+  }
+
+  @override
+  void visitIfElement(IfElement node) {
+    if (node.thenElement is AdjacentStrings) {
+      rule.reportLint(node.thenElement);
```

Comment:
We probably don't want to report a lint here if there's an `elseElement`, but should report a lint in the `elseElement` if it exists. For example, given
```
var list = [
  if (c) 'a' 'b' else 'c' 'd'
];
```
It wouldn't make any sense to add a comma between `'a'` and `'b'` because `'b' else 'c' 'd'` isn't be a valid expression. But it would name sense to suggest a comma between `'c'` and `'d'`.

Reply:
Great catch! Done.